---
output:
  html_document: default
  pdf_document: default
---

This is an html rendering of the repository (https://github.com/jeffreyblanchard/GenomeAnalysisR) for the course Human Genome Analysis in R taught at the University of Massachusetts Amherst in Spring 2020 by Prof. Jeffrey Blanchard

## Labs

### R Basics

* 1 : [Introduction to R and Reproducible Research](Lab1_HuGen_2020.html)
* 2 : [Basic R Object types](Lab2_HuGen_2020.html) and [Producing Reports With knitr](Lab2B_HuGen_2020.html)

### The R Tidyverse

* 3 : [Graphics with ggplot2](Lab3_HuGen_2020.html)
* 4 : [Publication Graphics](Lab4_HuGen_2020.html)
* 5 : [Data Wrangling](Lab5_HuGen_2020.html)
* 6 : [Github and Collaborative Projects](Lab6_HuGen_2020.html)

### RNA-seq Analysis

* 7 : [Data Preprocessing and Normalization](Lab7_HuGen_2020.html)
* i : Data Exploration
* ii : Differential Expression and Box Plots
* iii : Genome Visualization
* iv : Gene Ontology Analysis
* v : Network Analysis

### Genealogies

* 8 : [Genealogical Trees](Lab8_HuGen_2020.html)

### COVID-19 reporting data and SARS-CoV-2 genome analysis

* 9 : [Working with the COVID-19 data from GitHub](Lab9_HuGen_2020.html)
* 10 : [Data Maps, Interactive Graphs and Animations from the COVID-19 reporting data](Lab10_HuGen_2020.html)
* 11 : Genome mapping of SARS-CoV-2 variants
* 12 : Phylogenetic and Network Analysis of SARS-CoV-2



### Students taking 697 will be expected to complete these following additional labs

* 13 : [Project Organization and Management](https://datacarpentry.org/organization-genomics/01-tidiness/index.html)
* 14 : [The Command Line](LabA_697_command_line.html)
* 15 : [Accessing High Performance Computers](LabB_697_MGHPCC.html)
* 16 : [Assessing Read Quality](LabC_697_SNP_mapping.html)
* 17 : [Trimming and Filtering](LabC_697_SNP_mapping.html)
* 18 : [Read Mapping - Variant Calling Workflow](LabC_697_SNP_mapping.html)
* 19 : Read Mapping - Transcript Counts Workflow
